app/src/main/java/com/everest/userinfo/EditActivity.kt

```diff
@@ -0,0 +1,165 @@
+package com.everest.userinfo
+
+import android.content.Intent
+import android.graphics.Color
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.View
+import android.widget.EditText
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.everest.userinfo.databinding.ActivityEditBinding
+import java.util.regex.Pattern
+
+class EditActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityEditBinding
+    private var isEnabled: Boolean = false
+
+    private val EMAIL_ADDRESS_PATTERN: Pattern = Pattern.compile(
+        "[a-zA-Z0-9\\+\\.\\_\\%\\-\\+]{1,256}" + "\\@" + "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,64}" + "(" + "\\." + "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,25}" + ")+"
+    )
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityEditBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.validateButton.setOnClickListener {
+            validateData()
+        }
+
+        binding.confirmButton.setOnClickListener {
+            val intent = Intent(this, ConfirmActivity::class.java)
+            intent.putExtra(USERNAME, binding.usernameET.text.toString())
+            intent.putExtra(EMAIL, binding.emailET.text.toString())
+            intent.putExtra(PHONE_NUMBER, binding.phoneNumberET.text.toString())
+            intent.putExtra(PINCODE, binding.pincodeET.text.toString())
+            intent.putExtra(ADDRESS, binding.addressET.text.toString())
+            startActivity(intent)
+        }
+
+        binding.cancelButton.setOnClickListener {
+            editDetails(binding.layout)
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putInt(VALIDATE_BUTTON, binding.validateButton.visibility)
```

Comment:
We should start using data classes, instead of passing every single value https://www.vogella.com/tutorials/AndroidParcelable/article.html